tools/src/launcher/Main.hs

```diff
@@ -338,13 +346,31 @@ main =
                     loNodeLogConfig
                     (NodeData loNodePath realNodeArgs loNodeLogPath)
                     (NodeData wpath loWalletArgs loWalletLogPath)
-                    (UpdaterData
-                        loUpdaterPath loUpdaterArgs loUpdateWindowsRunner loUpdateArchive)
+                    (setUpdaterData lo)
                     loNodeTimeoutSec
                     loReportServer
                     loWalletLogging
                 logNotice "Finished clientScenario"
   where
+    setUpdaterData :: LauncherOptions -> UpdaterData
+    setUpdaterData LO {..} = case buildOS of 
+        OSX -> OSXUpdaterData OSXUD
+            { oudOpenPath = loUpdaterPath
+            , oudArgs = loUpdaterArgs
+            -- It would be better to use `bug` function from universum-1.0.1 but you want 0.9.0
+            -- , oudPkgPath = fromMaybe (bug Not_set_OSX_loUpdateArchive) loUpdateArchive
+            , oudPkgPath = fromJust loUpdateArchive
+            }
+        Windows -> WindowsUpdaterData WinUD
+            { wudExePath = loUpdaterPath
+            , wudArgs = loUpdaterArgs
+            -- It would be better to use `bug` function from universum-1.0.1 but you want 0.9.0
+            -- , wudBatPath = fromMaybe (bug Not_set_Windows_loUpdateWindowsRunner) loUpdateWindowsRunner
+            , wudBatPath = fromJust loUpdateWindowsRunner            
+            }
+        -- It would be better to use `bug` function from universum-1.0.1 but you want 0.9.0
+        _ -> UnsupportedOSUpdateData UnsupOSUD
```

Comment:
This line is not a bug, it's absolutely normal, there shouldn't be any errors, it's ok to run launcher on linux, for example.